x/merkledb/db.go

```diff
@@ -634,35 +654,31 @@ func (db *merkleDB) GetChangeProof(
 
 // NewView returns a new view on top of this trie.
 // Changes made to the view will only be reflected in the original trie if Commit is called.
-// Assumes [db.lock] isn't held.
-func (db *merkleDB) NewView() (TrieView, error) {
-	return db.NewPreallocatedView(defaultPreallocationSize)
+// Assumes [db.commitLock] isn't held.
+func (db *merkleDB) NewView(batchOps []database.BatchOp) (TrieView, error) {
+	db.commitLock.RLock()
+	defer db.commitLock.RUnlock()
+
+	newView, err := db.newUntrackedView(batchOps)
+	if err != nil {
+		return nil, err
+	}
+	db.childViews = append(db.childViews, newView)
```

Comment:
doesn't taking RLock here make it possible for 2 different appends to happen concurrently?